src/server/external-services/pathway-commons.js

```diff
@@ -121,23 +125,70 @@ const sifGraph = async ( queryObj ) => {
   });
 };
 
-/* xref2Uri
- * Light wrapper around pc metadata service for mapping Xrefs db, id to url
+const handleEntityUriResponse = text => {
+  if( !text ) return ''; // Go ahead so we can cache, but gotta throw to inform clients
```

Comment:
So this is caching empty responses for invalid inputs?

Reply:
I can't reproduce the case for justifying this. Removing...

Reply:
Now I found it again: http://www.pathwaycommons.org/pc2/get?uri=http%3A%2F%2Fidentifiers.org%2Freactome%2FR-HSA-69541

This pathway contains 3 xrefs with a similar `db` field:
```xml
	<bp:RelationshipXref rdf:ID="RelationshipXref_reactomehttp___www_reactome_org_biopax_64_48887_RelationshipXref1782">
		<bp:relationshipType rdf:resource="#RelationshipTypeVocabulary_ac7de6f2f302971b64781fc96cc97c86" />
		<bp:comment rdf:datatype = "http://www.w3.org/2001/XMLSchema#string">REPLACED http://www.reactome.org/biopax/64/48887#RelationshipXref1782</bp:comment>
		<bp:id rdf:datatype = "http://www.w3.org/2001/XMLSchema#string">6804946</bp:id>
		<bp:db rdf:datatype = "http://www.w3.org/2001/XMLSchema#string">reactome database id release 64</bp:db>
	</bp:RelationshipXref>
```

I was caching this because I was concerned about how often these types of things make their way into BioPAX. Also I was  assuming this function wouldn't be exposed as an endpoint, so everything that comes in should be OK.

Reply:
simply ignore all the xrefs having db="reactome database id release 64" (or starts with "reactome database id release"), because these are internal and not useful xrefs, and there is always a standard stable xref with db="reactome" and id like "R-HSA-123456"...

Reply:
how does having a db field with 'reactome database id release 64' hit the empty case for this function?

Reply:
The fetch body.text() will be empty, but return HTTP 200.

Reply:
oh because reactome database id doesnt have a mapping?

Reply:
well I think what happening here is that Reactome is 'abusing' this Xref and basically putting comments rather than a valid db, id.

Reply:
If we get an empty response, we should consider throwing an error.

Reply:
Latest commit I let those empty responses throw up to caller.  Do not get cached.